File: node/src/controller/RecordController.ts
import {Request, Response} from 'express';
import {Autowire, ExpressController, GetMapping, PostMapping} from '@ppirogov/express-decorators';
import {RecordServiceInterface} from '../service';

@ExpressController
export class RecordController {

  @Autowire('RecordService')
  recordService: RecordServiceInterface;

  @PostMapping('/api/records')
  recordPost(req: Request, res: Response) {
    this.recordService.createRecord(req.body).then(record => {
      res.setHeader('Content-Type', 'application/json');
      res.send(record);
    }).catch(error => {
      console.error(error);
      res.status(500).end();
    });
  }

  @GetMapping('/api/records')
  recordsGet(req: Request, res: Response) {
    this.recordService.getRecords().then(records => {
      res.setHeader('Content-Type', 'application/json');
      res.send(records);
    }).catch(error => {
      console.error(error);
      res.status(500).end();
    });
  }

}

File: spring/src/main/java/com/record/api/controller/RecordController.java
package com.record.api.controller;

import com.record.persistence.entity.Record;
import com.record.service.RecordService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class RecordController {

    @Autowired
    RecordService recordService;

    @GetMapping(value = "/api/records", produces = "application/json")
    public Iterable<Record> recordsGet() {
        return this.recordService.getRecords();
    }

    @PostMapping(value = "/api/records", produces = "application/json")
    public Record recordPost(@RequestBody Record item) {
        return this.recordService.createRecord(item);
    }
}
